### DISCUSSION TITLE
Should canDrop: () => false also block dropping on children?

### DISCUSSION THREAD

**Discussion by alexreardon** [atlassian/pragmatic-drag-and-drop#45] [Ideas]
Started from: #44 

Right now each drop target can specify whether it can be dropped on with `canDrop`. `canDrop: () => false` will not block dropping on parents or children. Some people might expect that a disabled parent should disable children (eg like a how a disabled `<fieldset>` element disables internal form elements).

A few potential options:

- Make it so that a disabled parent always disables all children
- Make it so a disabled parent disables children, unless a child states it wants to be disabled. This is problematic as the default `canDrop()` behaviour is `true`. I don't think we should distinguish between explicit `canDrop: () => true` and a default `canDrop: () => true` as that adds extra hidden meaning to the API
- **[current state]** Keep the current behaviour (most flexible, but perhaps a bit of friction in that folks might think that a disabled parent should disable children).

I lean towards keep the **current state** for maximum flexibility, but I fully accept I might be off about this one. Keen to know what others think

**Comment by sami616** [comment]
I’m currently struggling with this one! My brain can’t quite figure out how best to disable all my child droppables.

 I tried storing location.current.dropTargets inside of a react ref from the onDropTargetsChanged callback, then grab the source from canDrop and just check if the source element exists in the first drop target element via element.contains()

Would love to see an example of how this might be done.
